Introduce manually entered queries in UI

epugh opened the pull request:
### Description
Previously the only way to enter a query was to make a JSON file, and skip the reference Answer, or sample via UBI.

This lets you just enter a set of queries.  It's built around the "My boss sent me a list of problem queries, and I want to make a query set" out of it use case.

This is a replacement for https://github.com/opensearch-project/dashboards-search-relevance/pull/548 that adds in unit testing, and matches the new structure for the query set UI code.

### Issues Resolved
n/a

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

epugh committed:
Introduce manually entered queries

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
Wordsmith the help text for how many queries in a query set

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh commented:
@fen-qin  can I get a review on this?   I'd really like to get it in on Monday.....

heemin32 commented:
I believe both importing files and manual input have their advantages. Importing files is helpful when sharing data between users, while manual input is convenient for simple testing or experimenting with the feature.

epugh commented:
> I believe both importing files and manual input have their advantages. Importing files is helpful when sharing data between users, while manual input is convenient for simple testing or experimenting with the feature.

Exactly.   Teams that are working relevancy problems are constantly coming up with small sets of queries, say 7 to 20...  that they then are evaluation.   Managers and bosses constantly show up with "Why isn't my query I just typed in working?" and you want to bring up SRW and do a bit of quick triage and understanding the scope of the problem.

epugh committed:
Merge remote-tracking branch 'upstream/main' into simplify_entering_queries_part_deux

epugh committed:
Refactor the processQueryFile into two seperate methods to remove passing around isPlainText property.

We've successfully refactored `processQueryFile` to split it into two separate methods:
   - `processQueryFile`: Handles JSON format files
   - `processPlainTextFile`: Handles plain text files where each line is a query

Added tests for `processPlainTextFile`
   - Updated the mocks in `use_query_set_form.test.ts` to handle both functions

epugh committed:
Add Changelog entry.

epugh committed:
Additional tests

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
Add more tests

epugh committed:
From User perspective, the query preview doesn't care if you are typing it in, or uploading a file.  So simplify.

epugh commented:
I wanted to follow up on this?  Could I get a review?  Hoping to close out some of these tickets that came up in prep for OpenSearch Con.

heemin32 commented:
Could you increase the test coverage and fix the build failure?

epugh committed:
Merge remote-tracking branch 'upstream/main' into simplify_entering_queries_part_deux

epugh committed:
Improve test coverage for query_set hooks

epugh committed:
Changes suggested by the same script in dashboards-reporting project

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
Swap from compiling OpenSearch to using released version

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
Try known existing location for 3.2.0.0 version of plugin

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
fix yaml syntax

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
remove security policy which causes problems because we dont grant those permissions

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
use non versioned local name

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
get paths right

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
more streamlined approach of just one file

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
filter param is apparently unknown/not supported anymore

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
follow pattern in reporting dashboard

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
try firing up cypress the way dashboards-observability does it

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh commented:
Okay, I am running out of steam on this..   I can run the cypress tests, and they basically report that the OSD isn't starting up.

epugh commented:
@heemin32 could we move the fixing of the end to end tests to a seperate piece of work?  I noticed that both dashboards-reports and dashboards-observability have failing Cypress tests.    Since they are what I am modeling trying to fix ours, and if they are failing, then that doesn't give me much hope of it passing.   I've spent most of the day on this, so I need to take a break, do you have some suggestions on where to look?  Maybe another dashboard project where the cypress tests ARE running?   Or, let this one get merged, and I'll open a seperate PR focused on fixing the test?

fen-qin commented:
I tried to check the E2E workbench test results
```
    Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  plugins/search-relevance-dashboards      01:00        1        -        1        -        - │
  │    /1_query_compare.spec.js                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  plugins/search-relevance-dashboards        1ms        -        -        -        -        - │
  │    /2_search_card.spec.js                                                                      │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  plugins/search-relevance-dashboards      01:06        5        -        1        -        4 │
  │    /3_query_set_creation.spec.js                                                               │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  plugins/search-relevance-dashboards      01:05        4        -        1        -        3 │
  │    /4_search_configuration_creation.sp                                                         │
  │    ec.js                                                                                       │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  plugins/search-relevance-dashboards      01:05        4        -        1        -        3 │
  │    /5_judgment_creation.spec.js                                                                │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  plugins/search-relevance-dashboards      01:05        6        -        1        -        5 │
  │    /6_experiment_creation.spec.js                                                              │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  5 of 6 failed (83%)                      05:23       20        -        5        -       15  
```

And here is one of the video recordings for the query set creation. Seems like the it failed to load baseUrl. 

https://github.com/user-attachments/assets/dac77947-6916-4d97-855f-b4207e48ee9b

fen-qin committed:
update e2e workflow os version to 3.3 (#640)

Signed-off-by: Fen Qin <mfenqin@amazon.com>

epugh committed:
Back out edits made

epugh commented:
@arunagov, remember how hard it was to add queries to SRW, this ticket will make it easy, you jsut type them in.

heemin32 commented:
@epugh Any update on the PR?

epugh committed:
Merge remote-tracking branch 'upstream/main' into simplify_entering_queries_part_deux

epugh committed:
update and restore test

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh commented on public/components/query_set/utils/file_processor.ts:
```diff
@@ -13,14 +13,48 @@ export interface FileProcessResult {
+      // Each line is a query in plain text mode
+      queryList.push({
+        queryText: line.trim(),
+        referenceAnswer: '',
```
I agree..   so, I think I have to change things up a bit now that referenceAnswer is optional, so we aren't passing `''` around all over the place.

epugh committed:
Removes the use of referenceAnswer: '' as a workaround by fixing the QueryItem interface.

Signed-off-by: Eric Pugh <epugh@opensourceconnections.com>

epugh committed:
helpText is not a supported property of EuiFilePicker, and throws a console error

epugh committed:
Very light validation that manual input is NOT json formatted.

I tried some richer validations, but kept having edge case issues..  This works well enough.

epugh commented:
@fen-qin here is the new validation on JSON: 
<img width="685" height="558" alt="image" src="https://github.com/user-attachments/assets/519561fd-51b4-40c0-916c-582ea246bda2" />

epugh commented:
> > @fen-qin @heemin32 I believe I've addressed all the feedback and assuming tests pass, this is ready for merging!
> 
> @epugh, I still see there are three unresolved comments from @fen-qin.

Looking now at them.
